src/agent/centralized_configuration/include/centralized_configuration.hpp

```diff
@@ -38,21 +41,45 @@ namespace centralized_configuration
         /// @param getGroupIdFunction A function to get group IDs.
         void GetGroupIdFunction(GetGroupIdFunctionType getGroupIdFunction);
 
+        /// @brief Sets the function to save group IDs.
+        /// @details The "set-group" commands requires such function to be set.
+        /// @param saveGroupIdFunction A function to save group IDs.
+        void SaveGroupIdFunction(SaveGroupIdFunctionType saveGroupIdFunction);
```

Comment:
Maybe this function is not needed if we can do this https://github.com/wazuh/wazuh-agent/pull/274/files#r1863504509